lambdas/dla-migration/src/index.ts

```diff
@@ -1,9 +1,162 @@
+  if (has && shouldHave) {
+    return S3Path;
+  }
+  if (!has && !shouldHave) {
+    return S3Path;
+  }
+  if (!has && shouldHave) {
+    return `${S3Path}/`;
+  }
+  if (has && !shouldHave) {
+    let out = S3Path.slice(0, -1);
+    while (out.endsWith('/')) {
+      out = out.slice(0, -1);
+    }
+    return out;
+  }
+  /* this is just to satisfy typescript, it shouldn't be possible to get here */
+  return S3Path;
+};
+
+/**
+ * identifies whether the innermost folder of the filePath appears to be a timestamp
+ *
+ * @param targetPath
+ * @returns whether the innermost folder of the filePath appears to be a timestamp
+ */
+export const identifyDatedPath = (targetPath: string): boolean => (
+  moment(path.basename(path.dirname(targetPath)), 'YYYY-MM-DD').isValid()
+);
+
+/**
+ * Manipulate target filepath to put file is YYYY-MM-DD sub-folder
+ *
+ * @param targetPath
+ * @param message - DLARecord message from which timestamp can be extracted
+ * @returns targetPath with lowest 'folder' added in form YYYY-MM-DD
+ */
+export const addDateIdentifierToPath = (targetPath: string, message: DLARecord): string => (
+  path.join(
+    path.dirname(targetPath),
+    extractDateString(message),
+    path.basename(targetPath)
+  )
+);
+
+/**
+ * pull S3 Object from sourcePath, update it to new DLA structure and push it to targetPath
+ * noop if skip is true and an object already exists at targetPath
+ *
+ * @param bucket
+ * @param sourcePath
+ * @param targetPath
+ * @param skip - skip if targetPath already exists
+ * @returns whether the logic was actually run (not skipped)
+ */
+export const updateDLAFile = async (
+  bucket: string,
+  sourcePath: string,
+  targetPath: string
+): Promise<boolean> => {
+  const inDateForm = identifyDatedPath(targetPath);
+
+  logger.info(`About to process ${sourcePath}`);
+  const dlaObject = await getJsonS3Object(bucket, sourcePath);
+  const hoisted = await hoistCumulusMessageDetails(dlaObject);
+  let massagedTargetPath: string;
+  if (!inDateForm) {
+    massagedTargetPath = addDateIdentifierToPath(targetPath, hoisted);
+  } else {
+    massagedTargetPath = targetPath;
+  }
+
+  await putJsonS3Object(bucket, massagedTargetPath, hoisted);
+  logger.info(`Migrated file from bucket ${bucket}/${sourcePath} to ${massagedTargetPath}`);
+  if (massagedTargetPath !== sourcePath) {
+    await deleteS3Object(bucket, sourcePath);
+    logger.info(`Deleted file ${bucket}/${sourcePath}`);
+  }
+  return true;
+};
+
+/**
+ * update a batch of DLA files under prefix and push them to the targetDirectory
+ * skip files that appear to already have been processed if skip is set to true
+ *
+ * @param bucket
+ * @param targetDirectory
+ * @param prefix
+ * @param skip - skip files that are already present at the target directory, default false
+ */
+export const updateDLABatch = async (
+  bucket: string,
+  targetDirectory: string,
+  sourceDirectory: string
+): Promise<Array<boolean>> => {
+  const out = [];
```

Comment:
We can just add trailing /:
s3Path.replace(//?$/, '/')

Reply:
you're right, this is a leftover from more complex logic that needed to do a lot more iwth this